checker/src/main/java/org/checkerframework/checker/mustcall/MustCallTransfer.java

```diff
@@ -0,0 +1,253 @@
+      TernaryExpressionNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitTernaryExpression(node, input);
+    updateStoreWithTempVar(result, node);
+    return result;
+  }
+
+  /**
+   * This method either creates or looks up the temp var t for node, and then updates the store to
+   * give t the same type as node.
+   *
+   * @param node the node to be assigned to a temporary variable
+   * @param result the transfer result containing the store to be modified
+   */
+  public void updateStoreWithTempVar(TransferResult<CFValue, CFStore> result, Node node) {
+    // Must-call obligations on primitives are not supported.
+    if (!TypesUtils.isPrimitiveOrBoxed(node.getType())) {
+      LocalVariableNode temp = getOrCreateTempVar(node);
+      if (temp != null) {
+        JavaExpression localExp = JavaExpression.fromNode(temp);
+        AnnotationMirror anm =
+            atypeFactory
+                .getAnnotatedType(node.getTree())
+                .getAnnotationInHierarchy(atypeFactory.TOP);
+        insertIntoStores(result, localExp, anm == null ? atypeFactory.TOP : anm);
+      }
+    }
+  }
+
+  /**
+   * Either returns the temporary variable associated with node, or creates one if one does not
+   * exist.
+   *
+   * @param node a node, which must be an expression (not a statement)
+   * @return a temporary variable node representing {@code node} that can be placed into a store
+   */
+  private @Nullable LocalVariableNode getOrCreateTempVar(Node node) {
+    LocalVariableNode localVariableNode = atypeFactory.tempVars.get(node.getTree());
+    if (localVariableNode == null) {
+      VariableTree temp = createTemporaryVar(node);
+      if (temp != null) {
+        IdentifierTree identifierTree = treeBuilder.buildVariableUse(temp);
+        localVariableNode = new LocalVariableNode(identifierTree);
+        localVariableNode.setInSource(true);
+        atypeFactory.tempVars.put(node.getTree(), localVariableNode);
+      }
+    }
+    return localVariableNode;
+  }
+
+  /**
+   * Creates a variable declaration for the given expression node, if possible.
+   *
+   * @param node an expression node
+   * @return a variable tree for the node, or null if an appropriate containing element cannot be
+   *     located
+   */
+  protected @Nullable VariableTree createTemporaryVar(Node node) {
+    ExpressionTree tree = (ExpressionTree) node.getTree();
+    TypeMirror treeType = TreeUtils.typeOf(tree);
+    Element enclosingElement;
+    TreePath path = atypeFactory.getPath(tree);
+    if (path == null) {
+      enclosingElement = TreeUtils.elementFromTree(tree).getEnclosingElement();
+    } else {
+      ClassTree classTree = TreePathUtil.enclosingClass(path);
+      enclosingElement = TreeUtils.elementFromTree(classTree);
+    }
+    if (enclosingElement == null) {
+      return null;
+    }
+    // Declare and initialize a new, unique iterator variable
```

Comment:
Why is this an iterator variable?  Are temporary variables only created for iterators and never for any other purpose?

Reply:
I think the comment was in error. I suspect this code was copy-pasted from code that created an iterator variable originally, and we forgot to delete this comment. I've removed it, since it was non-sensical.